### ISSUE TITLE
[Weird question] Does Tensorflow send data to Google?

### ISSUE THREAD

**Issue by Darthagnon** [photoprism/photoprism#598]
Bit of a noob myself, and I was wondering how Tensorflow is implemented? Is it an external Google API call, sending data to them, or is it based off a static local pretrained AI model binary? 

I'm guessing it's used for all search functions - does it change much if disabled?

**Comment by bencark** [comment]
i believe it's the latter

**Comment by lastzero** [comment]
We're not using any Google APIs. We use a local TensorFlow model for this.

You won't see any (automatically created) labels when TensorFlow is disabled. NSFW detection also won't work.

**Comment by Darthagnon** [comment]
Thank you so much, lastzero!

Just checking, is the Tensorflow model included in the docker image?

**Comment by lastzero** [comment]
Yes

**Comment by sbrl** [comment]
I'm a Tensorflow user myself for another project. Tensorflow does NOT send any data to Google. If, however, you were to build Photoprism from source, you would send some information about your computer to wherever you download Tensorflow from I would guess - but only one time.
